lib/rds-snapshot-export-pipeline-stack.ts

```diff
@@ -0,0 +1,185 @@
+                `${bucket.bucketArn}/*`,
+              ],
+              "Effect": "Allow"
+            }
+          ],
+        })
+      }
+    });
+
+    const lambdaExecutionRole = new Role(this, "RdsSnapshotExporterLambdaExecutionRole", {
+      assumedBy: new ServicePrincipal("lambda.amazonaws.com"),
+      description: 'RdsSnapshotExportToS3 Lambda execution role for the "' + props.dbName + '" database.',
+      inlinePolicies: {
+        "SnapshotExporterLambdaPolicy": PolicyDocument.fromJson({
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Action": "rds:StartExportTask",
+              "Resource": "*",
+              "Effect": "Allow",
+            },
+            {
+              "Action": "iam:PassRole",
+              "Resource": [snapshotExportTaskRole.roleArn],
+              "Effect": "Allow",
+            }
+          ]
+        })
+      },
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName("service-role/AWSLambdaBasicExecutionRole"),
+      ],
+    });
+
+    const snapshotExportEncryptionKey = new Key(this, "SnapshotExportEncryptionKey", {
+      alias: props.dbName + "-snapshot-exports",
+      policy: PolicyDocument.fromJson({
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Principal": {
+              "AWS": [
+                (new AccountRootPrincipal()).arn,
+                lambdaExecutionRole.roleArn
+              ]
+            },
+            "Action": [
+              "kms:Encrypt",
+              "kms:Decrypt",
+              "kms:ReEncrypt*",
+              "kms:GenerateDataKey*",
+              "kms:DescribeKey"
+            ],
+            "Resource": "*",
+            "Effect": "Allow",
+          },
+          {
+            "Principal": { "AWS": lambdaExecutionRole.roleArn },
+            "Action": [
+              "kms:CreateGrant",
+              "kms:ListGrants",
+              "kms:RevokeGrant"
+            ],
+            "Resource": "*",
+            "Condition": {
+                "Bool": {"kms:GrantIsForAWSResource": true}
+            },
+            "Effect": "Allow",
+          },
+          {
+            "Principal": { "AWS": (new AccountRootPrincipal()).arn },
+            "Action": [
+              "kms:CancelKeyDeletion",
+              "kms:Create*",
+              "kms:Delete*",
+              "kms:Describe*",
+              "kms:Disable*",
+              "kms:Enable*",
+              "kms:GenerateDataKey",
+              "kms:Get*",
+              "kms:List*",
+              "kms:Put*",
+              "kms:Revoke*",
+              "kms:ScheduleKeyDeletion",
+              "kms:TagResource",
+              "kms:UntagResource",
+              "kms:Update*"
+            ],
+            "Resource": "*",
+            "Effect": "Allow",
+          }
+        ]
+      })
+    });
+
+    const lambdaFunction = new lambda.Function(this, "LambdaFunction", {
```

Comment:
Give the function a meaningful name so we can recognise what this does/

Reply:
fixed